### PR TITLE
New plugin - fullscreen button

### PR DESCRIPTION
**Pull request by jon-heard**

### CODE REVIEW

File: src/fullscreen button.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+3. Click the transparent button and note that the game goes fullscreen.
+4. Hit the "escape" key and the game becomes non-fullscreen.
+*/
+
+const playerUi = document.getElementById('player');
+const fullscreenButton = document.createElement('button');
+playerUi.append(fullscreenButton);
+fullscreenButton.id = 'btnToggleFullscreen';
+fullscreenButton.setAttribute('style', 'position:absolute;width:21px;height:21px;right:2px;bottom:2px;padding:0;opacity:40%;');
+fullscreenButton.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" stroke="currentColor" stroke-linecap="round" fill="none" stroke-width="1.25" width="14" height="14"><path d="M5 2h-3v3"></path><path d="M11 2h3v3"></path><path d="M2 11v3h3"></path><path d="M14 11v3h-3"></path></svg>`;
```
Comment (seleb) [inline src/fullscreen button.js]: imo the icon + style here should be configurable, and the way you've styled this leaves most of the default styling intact, which will make it appear differently on different browsers. i'd recommend injecting a stylesheet and including more deliberate styles (including hover/focus state)

File: src/fullscreen button.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+HOW TO USE:
+1. Import this plugin into your game.
+2. Playtest the game.  Note the transparent button on the lower-right of the game window.
+3. Click the transparent button and note that the game goes fullscreen.
+4. Hit the "escape" key and the game becomes non-fullscreen.
+*/
+
+const playerUi = document.getElementById('player');
+const fullscreenButton = document.createElement('button');
```
Comment (seleb) [inline src/fullscreen button.js:43]: should add a `title` to this button: atm it has no accessible label

File: src/fullscreen button.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
+/**
+🖥️
+@file fullscreen button
+@summary Adds a fullscreen button to the lower-right of the game canvas.
+@license MIT
+@author Violgamba (Jon Heard)
+@version 4.7.0
```
Comment (seleb) [inline src/fullscreen button.js]:
```suggestion
```

**Review by seleb — CHANGES_REQUESTED** [review]
couple minor issues

**Comment by jon-heard** [comment]
I've made all suggested modifications.  I BELIEVE I've overridden all platform-specific button styling: size, padding, margin, color, border.

I tried really hard to show the button while in fullscreen mode.  It requires fullscreening the parent div of the canvas, instead of the canvas itself.  Unfortunately, the canvas parent has a transform style and when it's fullscreened that transform style is invalidated which messes with the fullscreen button size in a way that would be complex to fix.

**Comment by seleb** [comment]
> I tried really hard to show the button while in fullscreen mode. It requires fullscreening the parent div of the canvas, instead of the canvas itself. Unfortunately, the canvas parent has a transform style and when it's fullscreened that transform style is invalidated which messes with the fullscreen button size in a way that would be complex to fix.

imo it's better to not show the button in fullscreen mode for this case; the browser has native "exit fullscreen" UI that will be shown anyway so the user can't get stuck, and it's unlikely they'd want to toggle states unless they're done playing

**Review by seleb — APPROVED** [review]

**Pull request merged**